lib/puppet/indirector/resource/ral.rb

```diff
@@ -12,7 +12,7 @@ def allow_remote_requests?
   def find( request )
     # find by name
     res   = type(request).instances.find { |o| o.name == resource_name(request) }
-    res ||= type(request).new(:name => resource_name(request), :audit => type(request).properties.collect { |s| s.name })
+    res ||= type(request).new(:name => resource_name(request))
```

Comment:
This looks like it originally was designed to audit everything for the requested resource. I suspect that this is intended to allow the `ral` indirection to work for providers that do not implement `self.instances`

I don't know if we have any solid tests for the `ral` indirection (does the `puppet resource` command use it?) It's possible that this may break the indirection for some such types.

Digging through the history of this, it looks like audit replaced a previous mechanism for querying resource properties. We may need to re-implement such a system if resource querying was dependent on audit.